client/state/billing-data/util.js

```diff
@@ -0,0 +1,18 @@
+/**
+ * External dependencies
+ */
+import { assign } from 'lodash';
+import { moment } from 'i18n-calypso';
+
+/**
+ * Parses the date within a transaction.
+ * Returns a copy of the updated transaction.
+ *
+ * @param  {Object}  transaction  Transaction object
+ * @return {Object}               Updated transaction with parsed date
+ */
+export const parseDate = ( transaction ) => {
```

Comment:
Perhaps something more specific for the name? As is I would expect this to take in a string object and return a date.